lib/api_callable.js

```diff
@@ -168,6 +168,12 @@ function addTimeoutArg(aFunc, timeout, otherArgs, abTests) {
     options.deadline = new Date(now.getTime() + timeout);
     var metadata =
       otherArgs.metadataBuilder ? otherArgs.metadataBuilder(abTests) : null;
+    var headers = otherArgs.headers || {};
+    for (var key in headers) {
+      if (headers.hasOwnProperty(key)) {
+        metadata.set(key, headers[key]);
```

Comment:
This means to overwrite existing headers if the specified field collides with predefined items, like `method({"x-goog-api-client": "foo"})`.

I think the user-provided header info should be supplied beforehand, i.e.
1. create an instance of Metadata
2. set up fields specified from the user
3. add fields for gax use

I'm afraid that you need to modify metadataBuilder for this.